packages/scalecube-services/src/services/ServiceCall.js

```diff
@@ -29,28 +28,31 @@ export class ServiceCall {
                 }
                 throw Error("Message format error: data must be Array");
             })
-            .map(message => ({
-                message,
-                inst: this.router.route(message)
-            }))
-            .map((source) => {
-                if (source.inst && source.inst.service) {
-                    return source;
+            .map(message => {
+                const inst = this.router.route && this.router.route(message);
+                if (inst && inst.service) {
+                    return ({
+                        message,
+                        inst,
+                    })
                 }
                 throw Error(`Service not found error: ${message.serviceName}.${message.method}`);
             })
-            .map(source => ({
-                inst: source.inst,
-                message: source.message,
+            .map(({ inst, message }) => ({
+                inst,
+                message,
                 thisMs: this.microservices,
-                meta: source.inst.service.meta || source.inst.service.constructor.meta || {}
+                meta: inst.service.meta || inst.service.constructor.meta || {}
             }))
             .pipe(source$ => this.microservices.preRequest(source$))
             .map(obj => obj.inst)
             .switchMap(inst => utils.isLoader(inst) ?
                 Observable.from(new Promise(r => inst.service.promise.then(res => r(res)))) :
                 Observable.from([inst.service])
             )
+            .do((service) => {
+                this.microservices.postRequest({ service, message })
```

Comment:
@idanilt Do we need to pass some extra information in postRequest?

Reply:
I would use as the model model for postRequest
```typescript
                inst: serviceInstance,
                request: Message, // this called message on preRequest, need to change it
                response: Message,
                thisMs: this.microservices,
                meta: inst.service.meta || inst.service.constructor.meta || {}
```